### PR TITLE
Refactor/play + Role

### LINKED ISSUE #843: Role class + tests
Implementation:
Linear state machine of tactics, in the same way a tactic is a linear state machine of skills. Assigned to 1 robot. 

Testing:
Create some instances of tactics
Test if the progression through the tactic state machine is as expected.

### PR DESCRIPTION
**Pull request by Yuhanun**
Created two stub classes, feel free to throw suggestions

### CODE REVIEW

**Comment by Yuhanun** [comment]
Feel free to review.

File: include/roboteam_ai/stp/Tactic.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "stp/StpInfo.h"
+#include "stp/new_skills/Rotate.h"
+#include "stp/new_skills/SetDribbler.h"
+
+namespace rtt::ai::stp {
+    class Skill;
+
+    class Tactic {
+    private:
```
Comment (Ratoone) [inline include/roboteam_ai/stp/Tactic.h:19]: Protected. You want the derived classes to use or override these fields

File: include/roboteam_ai/stp/Tactic.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        /**
+         * This function should calculate any extra information that the skills might need to be executed.
+         * Things that should be calculated are for example how hard the kicker should shoot to get to the desired position
+         * or how fast the dribbler should be spinning.
+         *
+         * Though this method is responsible for ensuring everything is calculated, it helps to use helpers so this
+         * function doesn't become a massive hack
+         */
+        TacticInfo calculateInfoForSkill();
```
Comment (Ratoone) [inline include/roboteam_ai/stp/Tactic.h:30]: Provide TacticInfo and return SkillInfo - information for the skills contains the information for the tactic as well

File: include/roboteam_ai/stp/Tactic.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include "stp/StpInfo.h"
+#include "stp/new_skills/Rotate.h"
+#include "stp/new_skills/SetDribbler.h"
+
+namespace rtt::ai::stp {
+    class Skill;
+
+    class Tactic {
+    private:
+        rtt::collections::state_machine<Skill, Status, SkillInfo> skills{Rotate(), SetDribbler(), Rotate(), Rotate()};
```
Comment (Ratoone) [inline include/roboteam_ai/stp/Tactic.h:20]: Is initialization of skills required in the base class?

File: src/stp/Tactic.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// Created by jessevw on 03.03.20.
+//
+
+#include "include/roboteam_ai/stp/Tactic.h"
+#include "stp/StpInfo.h"
+
+namespace rtt::ai::stp {
+
+TacticInfo Tactic::calculateInfoForSkill() {
+}
```
Comment (Ratoone) [inline src/stp/Tactic.cpp:11]: Pure virtual or return empty struct

File: src/stp/new_skills/SetDribbler.cpp [hunk 5]
```diff
@@ -4,15 +4,14 @@
...
-
-void SetDribbler::onInitialize() { Skill::onInitialize(); }
-
-SetDribbler::Status SetDribbler::onUpdate() {
-  int dribblerSpeed = properties->getInt("dribblerSpeed");
+Status SetDribbler::onInitialize() noexcept {
+    return Status::Success;
+}
+//TODO implement this skill again
+Status SetDribbler::onUpdate(const rtt::ai::stp::SkillInfo &info) noexcept {
```
Comment (Ratoone) [inline src/stp/new_skills/SetDribbler.cpp:13]: Redundant namespace :P

File: include/roboteam_ai/stp/Tactic.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        virtual Status onUpdate(TacticInfo const &info) noexcept = 0;
+
+        virtual Status onTerminate() noexcept = 0;
+
+    public:
+        virtual Status initialize() noexcept;
+
+        virtual Status update(TacticInfo const &info) noexcept;
+
+        virtual Status terminate() noexcept;
```
Comment (rttwente) [inline include/roboteam_ai/stp/Tactic.h:43]:
```suggestion
        virtual void onInitialize() noexcept = 0;

        virtual Status onUpdate(TacticInfo const &info) noexcept = 0;

        virtual void onTerminate() noexcept = 0;

    public:
        virtual void initialize() noexcept;

        virtual Status update(TacticInfo const &info) noexcept;

        virtual void terminate() noexcept;
```

File: include/roboteam_ai/stp/new_skills/SetDribbler.h [hunk 7]
```diff
@@ -5,23 +5,22 @@
...
-     */
-    Status onUpdate() override;
-    void onTerminate(Status s) override;
+        Status onInitialize() noexcept override;
+        /**
+        * Sets the dribbler speed using blackboard parameter: "dribblerSpeed"
+        * @return status of the skill
+        */
+        Status onUpdate(SkillInfo const& info) noexcept override;
+        Status onTerminate() noexcept override;
```
Comment (rttwente) [inline include/roboteam_ai/stp/new_skills/SetDribbler.h]:
```suggestion
        void onInitialize() noexcept override;
        /**
        * Sets the dribbler speed using blackboard parameter: "dribblerSpeed"
        * @return status of the skill
        */
        Status onUpdate(SkillInfo const& info) noexcept override;
        void onTerminate() noexcept override;
```

File: include/roboteam_ai/stp/Tactic.h (hunk 1 above)
Comment (Yuhanun) [reply include/roboteam_ai/stp/Tactic.h:19]: not mine

**Review by RobotJente — APPROVED** [review]
Nice work! :)

**Pull request merged**